### ISSUE TITLE
PBIX instance, Error “A connection cannot be made. Ensure that the server is running.”

### ISSUE THREAD

**Issue by sherlockspreadsheets** [otykier/TabularEditor#405]
Sometimes Tabular Editor gets mixed up and cannot find the correct Port Number for the PBIX file. Sometimes I get the error, “**A connection cannot be made. Ensure that the server is running.**”.  

When I look at the computer processes for the msmdsrv.exe task, there are multiple Local Addresses and different Port Numbers. Why are there multiple port numbers, and how do I connect to the PBIX file when there are multiple ports?  Is there a way to manually enter the port number so Tabular Editor can use the correct port number, instead of the incorrect port number detected? Is it possible to make the PBIX file have only a single local port number?  

I don't have this problem all the time.  But it happens from time to time.  Tabular Editor cannot connect when this happens. 

![2020-03-17-Tabular Editor 2 9 2](https://user-images.githubusercontent.com/39740261/76895743-94bf4900-685e-11ea-857c-8af1f1fe56b1.png)

**Comment by otykier** [comment]
Power BI Desktop will use a new port number for each running instance - there is no way to control this, nor can we have multiple PBIX files share the same port number, as each instance of Desktop spins up its own instance of Analysis Services (msmdmsrv.exe).

I'm not able to reproduce the behaviour of Tabular Editor not picking up the correct port numbers. If a port number isn't showing, you can still connect by typing "localhost:xxxxx" in the "Server" field, where "xxxxx" is the port number you want to connect to.

**Comment by albertospelta** [comment]
I'm not able to reproduce but I guess the issue may be caused by this hack:

https://github.com/otykier/TabularEditor/blob/494c0198d581882d6d3165f9dc98eb5d6290c817/TabularEditor/UIServices/ManagedIpHelper.cs#L174-L176

**Comment by sherlockspreadsheets** [comment]
Thank you for your response, @albertospelta and @otykier.  I was able to manually enter the connection string as described, “localhost:xxxxx”.  Thanks for sharing this tip.  I'm not having the issue today.  I tried recreating it by restoring the problem file.  It was a composite model (CSV, SQLAZ), and I figured maybe that had something to do with it.  Alas, I was unable to recreate the issue.  The dropdown connection is working today.  If it happens again I will try manually entering the server connection string.
